plone/formwidget/namedfile/widget.py

```diff
@@ -64,6 +82,35 @@ class NamedFileWidget(Explicit, file.FileWidget):
     klass = u'named-file-widget'
     value = None  # don't default to a string
 
+    @property
+    def is_upload(self):
+        """True, if value is a uploaded file.
+        """
+        return isinstance(self.value, FileUpload)
+
+    @property
+    def file_upload_id(self):
+        """Store the uploaded file contents with a file_upload_id key in a
+        temporary folder.
+        This is done, so that in case of a form validation error the already
+        uploaded image can be reused.
+        """
+        if self.is_upload:
+            upload_id = uuid.uuid4().hex
+            tmp_folder = getSite().temp_folder  # Acquire from Zope root.
```

Comment:
Not sure, if usage of the temp folder is OK and if Acquisition from outside the site root is OK.
However, a temp folder is what I want for that.

Reply:
Alternatively, I could use an annotation on the Plone root. But the advantage with a temporary directory is, that not much cleanup is needed and it might be on a very efficient storage (e.g. RAM).

Reply:
But using the temp folder means it's local to a particular Zope instance. If the load balancer isn't sticky, the client won't necessarily reach the same instance on every request. I think I would go with a BTree on the portal root.